OpenRA.Mods.D2k/Traits/SpiceBloom.cs

```diff
@@ -78,6 +84,12 @@ public SpiceBloom(ActorInitializer init, SpiceBloomInfo info)
 
 		public void Tick(Actor self)
 		{
+			if (!self.World.Map.Contains(self.Location))
+				return;
+
+			if (info.GrowthTerrainTypes.Any() && !info.GrowthTerrainTypes.Contains(self.World.Map.GetTerrainInfo(self.Location).Type))
```

Comment:
Minor perf nit: Prefer `.Count > 0` since `Any` needs to allocate and this is called every tick.